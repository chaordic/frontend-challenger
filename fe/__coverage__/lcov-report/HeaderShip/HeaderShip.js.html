<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for HeaderShip/HeaderShip.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">HeaderShip</a> HeaderShip.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Fragment } from 'react';
import { object } from 'prop-types';
import cssModules from 'react-css-modules';
import styles from './styles.scss';
import ShipRelated from '../ShipRelated/ShipRelated';
&nbsp;
const HeaderShip = ({ order, fulfillments }) =&gt; (
  &lt;section className={styles.grid}&gt;
    {!order.isLoading ? (
      &lt;Fragment&gt;
        &lt;article&gt;
          &lt;h3 className={styles.title}&gt;Pedido&lt;/h3&gt;
          &lt;p className={styles.info}&gt;{order.id}&lt;/p&gt;
        &lt;/article&gt;
&nbsp;
        &lt;article&gt;
          &lt;h3 className={styles.title}&gt;Status&lt;/h3&gt;
          &lt;p className={`${styles.info} ${order.status &amp;&amp; styles[order.status.toLowerCase()]}`}&gt;{order.status}&lt;/p&gt;
        &lt;/article&gt;
      &lt;/Fragment&gt;
    ) : (
<span class="branch-1 cbranch-no" title="branch not covered" >      &lt;article className={styles.loading} /&gt;</span>
    )}
&nbsp;
    {fulfillments !== undefined &amp;&amp; fulfillments.items.length &gt; 0 &amp;&amp;
      &lt;article&gt;
        &lt;h3 className={styles.title}&gt;Entregas relacionadas&lt;/h3&gt;
        {fulfillments.items.map((e, i) =&gt; (
          &lt;ShipRelated id={e.id} key={i} /&gt;
        ))}
      &lt;/article&gt;
    }
&nbsp;
  &lt;/section&gt;
);
&nbsp;
HeaderShip.propTypes = {
  order: object.isRequired,
  fulfillments: object.isRequired,
};
&nbsp;
HeaderShip.defaultProps = {
};
&nbsp;
export default cssModules(HeaderShip, styles, { allowMultiple: true });
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 11 2019 23:02:12 GMT+0000 (WET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
