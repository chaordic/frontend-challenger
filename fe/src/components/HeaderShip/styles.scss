@import '../../global-styles/settings.global.scss';

.grid {
    display: grid;
    grid-template-columns: repeat(6, 16%);
    grid-column-gap: 10px;
    margin: 10px 20px;
}

.title {
    color: #798896;
    font-weight: 400;
    font-size: 16px;
    font-family: sans-serif;
    padding: 0 0 5px 0;
    text-align: left;
}

.info {
    text-align: left;
    color: #000;
}