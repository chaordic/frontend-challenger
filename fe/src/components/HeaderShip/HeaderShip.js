import React from 'react';
import { object } from 'prop-types';
import cssModules from 'react-css-modules';
import styles from './styles.scss';
import ShipRelated from '../ShipRelated/ShipRelated';

const HeaderShip = ({ order, fulfillments }) => (
  <section className={styles.grid}>
    <article>
      <h3 className={styles.title}>Pedido</h3>
      <p className={styles.info}>{order.id}</p>
    </article>

    <article>
      <h3 className={styles.title}>Status</h3>
      <p className={styles.info}>{order.status}</p>
    </article>

    {fulfillments !== undefined && fulfillments.items.length > 0 &&
      <article>
        <h3 className={styles.title}>Entregas relacionadas</h3>
        {fulfillments.items.map(e => (
          <ShipRelated id={e.id} />
        ))}
      </article>
    }

  </section>
);

HeaderShip.propTypes = {
  order: object.isRequired,
  fulfillments: object.isRequired,
};

HeaderShip.defaultProps = {
};

export default cssModules(HeaderShip, styles, { allowMultiple: true });
