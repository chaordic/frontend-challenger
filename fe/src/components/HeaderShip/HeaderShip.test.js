import React from 'react';
import renderer from 'react-test-renderer';
import HeaderShip from './HeaderShip';

const props = {
  order: {
    isLoading: false,
    id: '22071559',
    status: 'PENDENTE',
    pointOfSale: 'E-commerce',
    createdAt: '5 de Fevereiro de 2019, às 19h30',
  },
  fulfillments: {
    isLoading: false,
    items: [{
      id: 'F1',
      locationId: '359',
      shipment: {
        zip: '00000-000',
        city: 'São Paulo',
        state: 'SP',
        number: '333',
        status: 'ACTIVE',
        country: 'Brasil',
        address1: 'Rua Oscar Freire',
        name: 'Renato Pereira',
        telephone: {
          type: 'billing',
          number: '(11) 98376-6343',
        },
      },
      status: 'ENTREGUE',
      type: 'ENVIADO',
      createdAt: '02/02/2019',
      updatedAt: '02/02/2019',
      freightCosts: {
        totalTime: 11,
        totalPrice: 10,
        deliveryEstimatedDate: '2019-03-05T19:30:00.000Z',
      },
      processedAt: '02/03/2019',
      locationType: 'Own Store',
      items: [{
        sku: 'AR384675',
        quantity: 2,
        stockType: 'PHYSICAL',
        name: 'Tenis Coca Coca Loretto - Feminino',
        size: '39',
        color: 'Branco, Cinza',
        image: 'http://image.url',
        price: 'R$100,00',
        desc: 'Branco, Cinza, 39',
      }, {
        sku: 'AR384677',
        quantity: 1,
        stockType: 'PHYSICAL',
        name: 'Tênis New Balance ML501 - Masculino',
        size: '40',
        color: 'Preto',
        image: 'http://image.url',
        price: 'R$100,00',
        desc: 'Preto, 40',
      }],
      geral: {
        quantity: 3,
        items: 2,
        totalFreight: 'R$0,00',
        totalPrice: 'R$200,00',
        total: 'R$200,00',
      },
      open: true,
      formattedAddress: 'Rua Oscar Freire, 333 São Paulo - SP - 00000-000 - Brasil',
      freight: {
        deliveryEstimatedDate: '02/03/2019',
        totalPrice: 'R$10,00',
      },
    }, {
      id: 'F2',
      locationId: '567',
      shipment: {
        zip: '00000-000',
        city: 'São Paulo',
        state: 'SP',
        number: '333',
        status: 'ACTIVE',
        country: 'Brasil',
        address1: 'Rua Oscar Freire',
        name: 'Renato Pereira',
        telephone: {
          type: 'billing',
          number: '(11) 98376-6343',
        },
      },
      status: 'PENDENTE',
      type: 'ENVIADO',
      createdAt: '02/02/2019',
      updatedAt: '04/10/2019',
      freightCosts: {
        totalTime: 11,
        totalPrice: 10,
        deliveryEstimatedDate: '2019-04-05T19:30:00.000Z',
      },
      processedAt: '04/10/2019',
      locationType: 'Own Store',
      items: [{
        sku: 'TN35568798',
        quantity: 1,
        stockType: 'PHYSICAL',
        name: 'Tênis Nike Air Max LTD 3 - Masculino',
        size: '39',
        color: 'preto/cinza',
        image: 'http://image.url',
        acquisitionType: 'PURCHASE',
        price: 'R$100,00',
        desc: 'preto/cinza, 39',
      }, {
        sku: 'TC587879785',
        quantity: 1,
        stockType: 'PHYSICAL',
        name: 'Tênis Nike Shox R4 - Masculino',
        size: '40',
        color: 'preto/cinza',
        image: 'http://image.url',
        acquisitionType: 'PURCHASE',
        price: 'R$100,00',
        desc: 'preto/cinza, 40',
      }],
      geral: {
        quantity: 2,
        items: 2,
        totalFreight: 'R$0,00',
        totalPrice: 'R$200,00',
        total: 'R$200,00',
      },
      open: true,
      formattedAddress: 'Rua Oscar Freire, 333 São Paulo - SP - 00000-000 - Brasil',
      freight: {
        deliveryEstimatedDate: '05/04/2019',
        totalPrice: 'R$10,00',
      },
    }],
    shipPlaces: ['Rua Oscar Freire, 333 São Paulo - SP - 00000-000 - Brasil'],
  },
};

describe('(Component) HeaderShip', () => {
  it('should render an HeaderShip with result', () => {
    const tree = renderer.create(<HeaderShip {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
