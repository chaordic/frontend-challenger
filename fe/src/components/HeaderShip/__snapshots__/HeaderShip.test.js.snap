// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) HeaderShip should render an HeaderShip with result 1`] = `
<section>
  <article>
    <h3>
      Pedido
    </h3>
    <p>
      22071559
    </p>
  </article>
  <article>
    <h3>
      Status
    </h3>
    <p
      className="undefined undefined"
    >
      PENDENTE
    </p>
  </article>
  <article>
    <h3>
      Entregas relacionadas
    </h3>
    <a
      href="#shipItem-F1"
      onClick={[Function]}
    >
      F1
    </a>
    <a
      href="#shipItem-F2"
      onClick={[Function]}
    >
      F2
    </a>
  </article>
</section>
`;
